text/0000-rust-crate-ownership.md

```diff
@@ -0,0 +1,158 @@
+
+From time to time a team's plan for a crate may change: experiments may conclude, crates may need to be deprecated, or the team may decide to release something for wider usage.
+
+In general, teams should notify core@rust-lang.org when such a transition is being made.
+
+Any transition _away_ from "Intentional Artifact" requires an RFC.
+
+Any transition to "Intentional Artifact" should ideally be accompanied by an RFC, and an update to the team charter if there is one.
+
+Expatriation should basically _never_ occur anymore, but it also requires an RFC and core team approval in case it is really necessary. If a team wishes to stop working on a crate, they should deprecate it and encourage the community to fork it or build their own thing.
+
+If "transitively intentional" crates are being deprecated care should be taken to ensure security issues will still be handled.
+
+Transitions between the other types can be made at will since they explicitly and clearly state their lack of a strong stability/maintenance guarantee.
+
+
+## Applying this to existing crates
+
+An audit should be performed on all existing potentially "official" crates, collecting them in a list and roughly determining what their team and category should be.
+
+(We have a list with a preliminary audit already and plan to post it to this RFC as an example soon)
+
+Once we have this list, we can approach teams with lists of crates and request that they verify that the categorization is accurate. In the case of some crates this might take some time as the team may need to work out what their intentions are with a particular crate.
+
+Then, working with the teams, we make these changes to their documentation. We also make sure all crates have the appropriate `rust-lang/teamname` github owner, and remove personal accounts from the owners list.
+
+For crates that are in direct use by a lot of the wider community, if we end up categorizing them as anything other than "intentional artifact", there should be an attempt to announce this "change" to the community. While there was no formal commitment made in case of these crates, the vague sense of officialness may have made people believe there was, and we should at least try to rectify this so that people are not continually misled. Whether or not this needs to be done, and how, can be figured out by the individual teams.
+
+A large part of this work can be parallelized; and it does not need to occur all at once.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This is a lot of work, but as we move towards a more deliberately structured project, it is probably necessary work.
+
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The alternative here is mostly to continue as is. This will become increasingly untenable as we add more and more crates; with the constant danger of internal crates becoming accidental artifacts that the ecosystem depends on.
```

Comment:
I'd argue that there's another alternative. You could tell teams to make it clear what level of support they offer for their creates, without standardizing the process or involving the core team. Pros: Less bureaucracy. Cons: No standardization in levels of support across the project, which could make it less clear what category each crate is in. I'm not saying this is a good alternative, but is certainly a better one than the one provided here. In general, I'd suggest that the alternatives presented should be the strongest ones, so that they stimulate discussion during the review process.

Reply:
Eh, idk. I think "involving core" is not actually the main thrust of this, that's a minor part of this proposal that makes it possible for us to keep track (the proposal still stands without "involving core"). But I'll mention this.